python/gui/MaaS/static/common/js/domain.js

```diff
@@ -0,0 +1,206 @@
+            return;
+        }
+    }
+    newParentSelect.appendChild(option);
+}
+
+function addDomainChoicesOption(values) {
+    var select = document.getElementById('domainChoices');
+    for (var i = 0; i < values.length; i++) {
+        var option = document.createElement('option');
+        option.value = values[i].substring(4);
+        option.innerHTML = values[i];
+        insertOptionInOrder(option, select);
+    }
+}
+
+function controlSelectAdd() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections');
+    for (var i = choices.options.length - 1; i >=0; i--) {
+        var opt = choices.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+            choices.removeChild(opt);
+            insertOptionInOrder(opt, selected);
+        }
+    }
+}
+
+function controlSelectRemove() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections'),
+        i,
+        opt;
+    for (i = selected.options.length - 1; i >=0; i--) {
+        opt = selected.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+            selected.removeChild(opt);
+            insertOptionInOrder(opt, choices);
+        }
+    }
+}
+
+function controlSelectAll() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections'),
+        i,
+        opt;
+    for (i = choices.options.length - 1; i >= 0 ; i--) {
+        opt = choices.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+        }
+        choices.removeChild(opt);
+        insertOptionInOrder(opt, selected);
+    }
+}
+
+function controlSelectClear() {
+    var choices = document.getElementById('domainChoices'),
+        selected = document.getElementById('domainSelections'),
+        i,
+        opt;
+    for (i = selected.options.length - 1; i >= 0 ; i--) {
+        opt = selected.options[i];
+        if (opt.selected) {
+            opt.selected = false;
+        }
+        selected.removeChild(opt);
+        insertOptionInOrder(opt, choices);
+    }
+}
+
+function loadFabricDomain(event) {
+    var name = $("#fabric-selector").val(),
+        type = "catchment", //$("#fabric-type-selector").val(),
+        catLists = [document.getElementById('domainChoices'),
+                    document.getElementById('domainSelections')],
+        loadingOverDiv = document.getElementById('loadCatsOverlay'),
+        select,
+        l,
+        i;
+
+    catLists[0].style.display = "none";
+    loadingOverDiv.style.display = "block";
+
+    $("input[name=fabric]").val(name);
+
+    // Clear any existing <option> tags from within "domainChoices" <select>
+    for (l = 0; l < catLists.length; l++) {
+        select = catLists[l];
+        for (i = select.options.length - 1; i >= 0; i--) {
+            select.remove(i);
+        }
+    }
```

Comment:
Try something like:

```javascript
for (let catchmentSelect of catLists) {
    for (let optionIndex = catchmentSelect.options.length - 1; optionIndex >= 0; optionIndex--) {
        catchmentSelect.remove(optionIndex);
    }
}
```

I may just be easier to do `$("#domainChoices option, #domainSelections option").remove()`, though.